climpred/relative_entropy.py

```diff
@@ -0,0 +1,291 @@
+            sigma_x = xr.DataArray(np.cov(pc_x.T))
+
+            r, d, s = _relative_entropy_formula(sigma_b, sigma_x, mu_x, mu_b,
+                                                neofs)
+
+            rel_ent.T.loc['R', init][t] = r
+            rel_ent.T.loc['D', init][t] = d
+            rel_ent.T.loc['S', init][t] = s
+    return rel_ent
+
+
+def _shuffle(ds, dim='initialization'):
+    """Shuffle ensemble members to uninitialize the data."""
+    old_dim_range = ds[dim]
+    shuffled = ds.sel({dim: np.random.permutation(ds[dim])})
+    if isinstance(ds, xr.DataArray):
+        shuffled[dim] = old_dim_range
+    elif isinstance(ds, xr.Dataset):
+        shuffled = shuffled.assign({dim: old_dim_range})
+    shuffled = shuffled.sortby(dim)
+    return shuffled
+
+
+def bootstrap_relative_entropy(initialized, control_uninitialized, sig=95,
+                               bootstrap=100, curv=True, neofs=None,
+                               ntime=None):
+    """
+    Bootstrap relative entropy threshold.
+
+    Generates a random uninitialized initialization and calculates the relative
+    entropy. sig-th percentile determines threshold level.
+
+    Args:
+        initialized (xr.DataArray): initialized ensemble with dimensions
+                                    initialization, member, time, lon (x),
+                                    lat(y).
+        control_uninitialized (xr.DataArray): control distribution with
+                                              dimensions initialization,
+                                              member, time, lon (x), lat(y).
+        sig (int): significance level for threshold.
+        bootstrap (int): number of bootstrapping iterations.
+        neofs (int): number of EOFs to use. Default: initialized.member.size
+        ntime (int): number of timestep to calculate.
+                     Default: initialized.time.size.
+        curv (bool): if curvilinear grids are provided disables EOF weights.
+
+    Returns:
+        rel_ent (pd.DataFrame): relative entropy sig-th percentile threshold.
+
+    """
+    if neofs is None:
+        neofs = initialized.member.size
+    if ntime is None:
+        ntime = initialized.time.size
+
+    x = []
+    for _ in range(min(1, int(bootstrap / initialized.time.size))):
+        shuffled_control_uninitialized = _shuffle(control_uninitialized).isel(
+            initialization=slice(0,initialized.initialization.size),
+            member=slice(0,initialized.member.size))
+        ds_pseudo_rel_ent = compute_relative_entropy(
+            shuffled_control_uninitialized, control_uninitialized, neofs=neofs,
+            curv=curv, ntime=ntime)
+        x.append(ds_pseudo_rel_ent)
+    ds_pseudo_metric = pd.concat(x, ignore_index=True)
+    qsig = sig / 100
+    sig_level = ds_pseudo_metric.stack().apply(lambda g: np.quantile(g, q=qsig))
+    return sig_level
+
+
+def plot_relative_entropy(rel_ent, rel_ent_threshold=None, **kwargs):
```

Comment:
It might be good to move this to `graphics.py` (which you can create in this PR) and then have a keyword on `compute_relative_entropy` that says `plot=False` but will plot the results for a first glance look.

Reply:
then it should also contain a bootstrap flag. personally I like the separation into graphics and computation though